commands/cloudapi-v6/natgateway.go

```diff
@@ -274,21 +282,30 @@ func RunNatGatewayUpdate(c *core.CommandConfig) error {
 }
 
 func RunNatGatewayDelete(c *core.CommandConfig) error {
-	if err := utils.AskForConfirm(c.Stdin, c.Printer, "delete nat gateway"); err != nil {
-		return err
-	}
+	var resp *resources.Response
 	dcId := viper.GetString(core.GetFlagName(c.NS, cloudapiv6.ArgDataCenterId))
 	natGatewayId := viper.GetString(core.GetFlagName(c.NS, cloudapiv6.ArgNatGatewayId))
-	c.Printer.Verbose("NatGateway with id: %v is deleting...", natGatewayId)
-	resp, err := c.CloudApiV6Services.NatGateways().Delete(dcId, natGatewayId)
-	if resp != nil {
-		c.Printer.Verbose(cloudapiv6.RequestTimeMessage, resp.RequestTime)
-	}
-	if err != nil {
-		return err
-	}
-	if err = utils.WaitForRequest(c, waiter.RequestInterrogator, printer.GetId(resp)); err != nil {
-		return err
+	allFlag := viper.GetBool(core.GetFlagName(c.NS, cloudapiv6.ArgAll))
+	if allFlag {
+		err := DeleteAllNatgateways(c)
```

Comment:
Here, it needs to be returned also the `resp`, and not use ":=" to not define it as a new variable as it is already defined above and used outside the if loop.